contracts/src/set_verifier.rs

```diff
@@ -116,4 +133,110 @@ where
+    pub async fn get_verified_root_seal(&self, root: B256) -> Result<Bytes> {
+        self.query_verified_root_event(root, None, None).await
+    }
+
+    async fn get_latest_block(&self) -> Result<u64> {
+        self.instance
+            .provider()
+            .get_block_number()
+            .await
+            .context("Failed to get latest block number")
+    }
+
+    /// Query the VerifiedRoot event based on the root and block options.
+    /// For each iteration, we query a range of blocks.
+    /// If the event is not found, we move the range down and repeat until we find the event.
+    /// If the event is not found after the configured max iterations, we return an error.
+    /// The default range is set to 100 blocks for each iteration, and the default maximum number of
+    /// iterations is 100. This means that the search will cover a maximum of 10,000 blocks.
+    /// Optionally, you can specify a lower and upper bound to limit the search range.
+    async fn query_verified_root_event(
+        &self,
+        root: B256,
+        lower_bound: Option<u64>,
+        upper_bound: Option<u64>,
+    ) -> Result<Bytes> {
+        let mut upper_block = upper_bound.unwrap_or(self.get_latest_block().await?);
+        let start_block = lower_bound.unwrap_or(upper_block.saturating_sub(
+            self.event_query_config.block_range * self.event_query_config.max_iterations,
+        ));
+
+        // Loop to progressively search through blocks
+        for _ in 0..self.event_query_config.max_iterations {
+            // If the current end block is less than or equal to the starting block, stop searching
+            if upper_block <= start_block {
+                break;
+            }
+
+            // Calculate the block range to query: from [lower_block] to [upper_block]
+            let lower_block = upper_block.saturating_sub(self.event_query_config.block_range);
+
+            // Set up the event filter for the specified block range
+            let mut event_filter = self.instance.VerifiedRoot_filter();
+            event_filter.filter = event_filter
+                .filter
+                .topic1(root)
+                .from_block(lower_block)
+                .to_block(upper_block);
+
+            // Query the logs for the event
+            let logs = event_filter.query().await?;
+
+            // If we find a log, return the seal
+            if let Some((verified_root, _)) = logs.first() {
+                let seal = verified_root.seal.clone();
+                return Ok(seal);
+            }
+            // Move the upper_block down for the next iteration
+            upper_block = lower_block.saturating_sub(1);
+        }
+
+        // Return error if no logs are found after all iterations
+        bail!("VerifiedRoot event not found for root {:?}", root);
+    }
+
+    /// Decodes a seal into a [SetInclusionReceipt] including a [risc0_zkvm::Groth16Receipt] as its root.
+    pub async fn get_receipt(
```

Comment:
Same comment about `query` or `fetch`. Also similar comment about providing a based option that takes the (image_id, journal) and an "advanced" option that takes the receipt claim.
```suggestion
    pub async fn query_receipt(
```

Reply:
went with `fetch`